packages/manager/src/features/NodeBalancers/NodeBalancerSelect.test.tsx

```diff
@@ -0,0 +1,188 @@
+import { NodeBalancer } from '@linode/api-v4';
+import { screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { nodeBalancerFactory } from 'src/factories';
+import { renderWithTheme } from 'src/utilities/testHelpers';
+
+import { NodeBalancerSelect } from './NodeBalancerSelect';
+
+const fakeNodeBalancerData = nodeBalancerFactory.build({
+  id: 1,
+  label: 'metadata-test-region',
+  region: 'eu-west',
+});
+
+const TEXTFIELD_ID = 'textfield-input';
+
+describe('NodeBalancerSelect', () => {
+  test('renders custom options using renderOption', async () => {
+    // Create a mock renderOption function
+    const mockRenderOption = (
+      nodebalancer: NodeBalancer,
+      selected: boolean
+    ) => (
+      <span data-testid={`custom-option-${nodebalancer.id}`}>
+        {`${nodebalancer.label} - ${selected ? 'Selected' : 'Not Selected'}`}
+      </span>
+    );
+
+    // Render the component with the custom renderOption function
+    renderWithTheme(
+      <NodeBalancerSelect
+        multiple={false}
+        onSelectionChange={jest.fn()} // Placeholder, as there's no callback
+        options={[fakeNodeBalancerData]}
+        renderOption={mockRenderOption}
+        value={null}
+      />
+    );
+
+    const input = screen.getByTestId(TEXTFIELD_ID);
+
+    // Open the dropdown
+    userEvent.click(input);
+
+    // Wait for the options to load (use some unique identifier for the options)
+    await waitFor(() => {
+      const customOption = screen.getByTestId('custom-option-1');
+      expect(customOption).toBeInTheDocument();
+      expect(customOption).toHaveTextContent(
+        'metadata-test-region - Not Selected'
+      );
+    });
+  });
+  test('should display custom no options message', async () => {
```

Comment:
I think using `it` over `test` reads better
```suggestion
  it('should display custom no options message if one is passed in', async () => {
```